gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java

```diff
@@ -50,16 +50,14 @@
 
 
 /**
- * Helper class with functionality meant to be re-used between the DagManager and Orchestrator when launching
+ * Helper class with functionality meant to be re-used between the LaunchDagProc and Orchestrator when launching
  * executions of a flow spec. In the common case, the Orchestrator receives a flow to orchestrate, performs necessary
- * validations, and forwards the execution responsibility to the DagManager. The DagManager's responsibility is to
- * carry out any flow action requests. However, with launch executions now being stored in the DagActionStateStore, on
- * restart or leadership change the DagManager has to perform validations before executing any launch actions the
- * previous leader was unable to complete. Rather than duplicating the code or introducing a circular dependency between
- * the DagManager and Orchestrator, this class is utilized to store the common functionality. It is stateful,
+ * validations, and creates DagActions. The DagProcessingEngine's responsibility is to
+ * process out dag action requests. However, with launch executions now being stored in the DagActionStateStore, on
+ * restart, the LaunchDagProc has to perform validations before executing any launch actions the previous LaunchDagProc
```

Comment:
unable to complete, right? like ran into some exception. either LaunchDagAction is processed and removed or not processed. What do you mean by "not fully processed"

Reply:
sorry, it's a minor point.  I was just saying that it could be that no dag proc has yet even tried to process the dag action.  hence all we can claim is that the DA is not yet fully processed.  overall, NBD